crates/bevy_debug_overlays/src/lib.rs

```diff
@@ -0,0 +1,197 @@
+}
+
+impl Plugin for DebugOverlaysPlugin {
+    fn build(&self, app: &mut App) {
+        let wgpu_features = app.world.resource::<RenderDevice>().features();
+        if !wgpu_features.contains(WgpuFeatures::TIMESTAMP_QUERY) {
+            panic!("DebugOverlaysPlugin added but RenderPlugin::wgpu_settings did not contain WgpuFeatures::TIMESTAMP_QUERY.");
+        }
+
+        if env::var("GPU_LOCKED").unwrap_or_default() != "magic" {
+            panic!("GPU clocks were not locked. Unable to get stable profiling results. Please rerun via the locking script.")
+        }
+
+        app.add_systems(Startup, setup_gpu_time_overlay)
+            .add_systems(
+                Update,
+                draw_gpu_time_overlay.run_if(on_timer(self.ui_update_frequency)),
+            );
+    }
+}
+
+impl Default for DebugOverlaysPlugin {
+    fn default() -> Self {
+        Self {
+            ui_update_frequency: Duration::from_millis(100),
+        }
+    }
+}
+
+#[derive(Component)]
+struct GpuTimerLabelMarker;
+
+#[derive(Component)]
+struct GpuTimerDurationMarker;
+
+fn setup_gpu_time_overlay(mut commands: Commands, asset_server: Res<AssetServer>) {
+    commands
+        .spawn(NodeBundle {
+            style: Style {
+                size: Size::width(Val::Percent(100.0)),
+                align_items: AlignItems::Start,
+                ..default()
+            },
+            ..default()
+        })
+        .with_children(|parent| {
+            parent
+                .spawn(NodeBundle {
+                    background_color: Color::rgba(0.10, 0.10, 0.10, 0.8).into(),
+                    style: Style {
+                        padding: UiRect::all(Val::Px(8.0)),
+                        flex_direction: FlexDirection::Column,
+                        ..default()
+                    },
+                    ..default()
+                })
+                .with_children(|parent| {
+                    parent.spawn(TextBundle::from_sections([
+                        TextSection::new(
+                            "GPU Time",
+                            TextStyle {
+                                font: asset_server.load("fonts/FiraSans-Bold.ttf"),
+                                font_size: 16.0,
+                                color: Color::WHITE,
+                            },
+                        ),
+                        TextSection::new(
+                            " (ms)  Mean     Min      Max\n",
+                            TextStyle {
+                                font: asset_server.load("fonts/FiraMono-Medium.ttf"),
+                                font_size: 10.0,
+                                color: Color::WHITE,
+                            },
+                        ),
+                    ]));
+                    parent.spawn(NodeBundle::default()).with_children(|parent| {
+                        let style = TextStyle {
+                            font: asset_server.load("fonts/FiraMono-Medium.ttf"),
```

Comment:
We can use the default font now.

Reply:
I'd rather keep it explicit, as I also use the bold version of the font in the plugin, which I can't use the default font for. We don't have a way of specifying "default font, but bold".

Reply:
that means that for someone to use that plugin outside of Bevy repo, they'll have to have the fonts at the exact same path...

Reply:
As it is, they can't really, as they need the GPU clock-locking script I included. I also only figured out how to do that for NVIDIA, so it's not broadly usable. Combined with the fact that I don't display nested profiling spans (only a flat list), this is really only (currently) meant for internal use, to aid in testing performance of rendering PRs.

How do you want to move further with this PR? Should I make the design worse and use the default font everywhere, or keep it as is?

Reply:
My vote is to make the design worse and use the default font everywhere. The default font exists largely for use cases like this. If I have to choose between "looks slightly worse" and "works everywhere by default", I choose "works everywhere by default". If we find a way to improve the styling for all "debug ui use cases", then we can apply that here. Imo the hackery here just isn't worth it.

Reply:
Done.